tests/RobotFramework/tests/customizing/data_path_config.robot

```diff
@@ -0,0 +1,85 @@
+*** Comments ***
+#Command to execute:    robot -d \results --timestampoutputs --log data_path_config.html --report NONE --variable HOST:192.168.1.120 /thin-edge.io/tests/RobotFramework/customizing/data_path_config.robot
+
+
+*** Settings ***
+Resource            ../../resources/common.resource
+Library             ThinEdgeIO
+Library             Cumulocity
+Library             JSONLibrary
+Library             String
+Library             DebugLibrary
+
+Suite Setup         Custom Setup
+Suite Teardown      Get Logs
+
+Force Tags          theme:cli    theme:configuration
+
+
+*** Test Cases ***
+Validate updated data path used by tedge-agent
+    Stop Service    tedge-agent
+    Execute Command    sudo tedge-agent --init
+    Start Service    tedge-agent
+    Directory Should Exist    /var/test/file-transfer
+
+Validate updated data path used by c8y-firmware-plugin
+    Stop Service    c8y-firmware-plugin
+    Execute Command    sudo c8y-firmware-plugin --init
+    Directory Should Exist    /var/test/firmware
+    Directory Should Exist    /var/test/cache
+    Start Service    c8y-firmware-plugin
+    Bootstrap child device with firmware operation support
+    ${firmware_url}=    Upload firmware binary to Cumulocity
+    ${date_from}=    Get Unix Timestamp
+    Create c8y_Firmware operation    ${firmware_url}
+    ${op_id}    ${cache_key}=    Validate tedge firmware update request sent    ${date_from}
+    File Should Exist    /var/test/firmware/${op_id}
+    File Should Exist    /var/test/cache/${cache_key}
+
+
+*** Keywords ***
+Custom Setup
+    ${PARENT_SN}=    Setup
+    Set Suite Variable    $PARENT_SN
+    Set Suite Variable    $CHILD_SN    ${PARENT_SN}_child
+    Execute Command    sudo mkdir /var/test
+    Execute Command    sudo chown tedge:tedge /var/test
+    Execute Command    sudo tedge config set data.path /var/test
+
+Custom Teardown
+    Execute Command    sudo rm -rf /var/test
+    Execute Command    sudo tedge config unset data.path
+    Get Logs
+
+Bootstrap child device with firmware operation support
+    Execute Command    mkdir -p /etc/tedge/operations/c8y/${CHILD_SN}
+    Sleep    3s
+    Execute Command    touch /etc/tedge/operations/c8y/${CHILD_SN}/c8y_Firmware
+    Sleep    3s
+    Cumulocity.Device Should Exist    ${CHILD_SN}
```

Comment:
We can make the test deterministic by removing the above `sleep` statements. Isn't it `Cumulocity. Device Should Exist`  is having retries to check whether the device is already created. If not may be this needs support from the library.

Reply:
Actually `Device Should Exist` will work just fine there, as the external id is fine, it is just the name of the device which is not ok sometimes....But in your instance you don't really care about the name, just the external id.

Reply:
Also there are other checks which allow waiting for inventory fragment changes...Or are you referring to wait for a child device (which is technically a linked inventory and not a custom fragment)?

```
Cumulocity.Device Should Have Fragments
Cumulocity.Device Should Have Fragment Values
```

Reply:
Glad to know that APIs like `Device Should Have Fragments` also exist. But from what I understood, these APIs check for the desired state only once. What we'd need on top of these existing APIs, to avoid these sleeps, is a "timed waiting" semantic, (similar to the functionality offered by this (waiting)[https://pypi.org/project/waiting/] package), with which we can conditionally wait for an inventory change(device created, desired fragment added etc) with a max timeout value.